### PR TITLE
Revert some changes

### PR DESCRIPTION
**Pull request by rodrigc**
Remove unnecessary changes which were breaking tests

### CODE REVIEW

**Comment by sideeffect42** [comment]
Thanks for your pull request. We've pretty much made the same changes except that you convert everything in RemoteUserAuth to byte strings while I went for unicode strings.

**Comment by rodrigc** [comment]
@sideeffect42 RemoteUserAuth must by byte strings, and you must not convert them to unicode strings.
Please merge this into your branch

**Comment by rodrigc** [comment]
Anything that deals with the www layer must be in byte strings.  So some of your changes were OK, but you went too far.  We need to fix things properly without breaking other stuff.

**Comment by sideeffect42** [comment]
OK. So we'll use `bytes` then. I digged further into it and resolved my problems in a different way.

I'm going to merge your branch and then apply my patches on top of it, so that you can have a look at them.

**Comment by rodrigc** [comment]
Solving these bytes/unicode problems is hard work.  (I've solved hundreds of them in buildbot, but missed a few, like what you have found).
I'm looking forward to your new patches.

**Comment by sideeffect42** [comment]
@rodrigc Check them out 314521b .

File: master/buildbot/www/ldapuserinfo.py [hunk 1]
```diff
@@ -119,8 +122,10 @@ def getLdapInfo(x):
             pattern = self.groupMemberPattern % dict(dn=dn)
             res = self.search(c, self.groupBase, pattern,
                               attributes=[self.groupName])
-            infos['groups'] = [bytes2unicode(g) for g in flatten(
-                [group_infos['raw_attributes'][self.groupName] for group_infos in res])]
+            _groups = flatten(
+                [group_infos['raw_attributes'][self.groupName] for group_infos in res])
+            infos['groups'] = [g if not isinstance(g, bytes) else g.decode('utf-8')
```
Comment (sideeffect42) [inline master/buildbot/www/ldapuserinfo.py]: @rodrigc Shouldn't the `infos` dict consist of `bytes`, too? Or are unicode strings acceptable here?

**Comment by sideeffect42** [comment]
@rodrigc `r.response` is 
```python
[{'raw_dn': b'cn=admins,ou=groups,dc=example,dc=com', 'dn': 'cn=admins,ou=groups,dc=example,dc=com', 'raw_attributes': {'cn': [b'admins']}, 'attributes': {'cn': ['admins']}, 'type': 'searchResEntry'}, { ... more groups ... }]
```
So it is both.

**Comment by rodrigc** [comment]
Hmm, `c.response` is coming entirely from the `ldap3` library?
If that is the case, then buildbot code needs to deal with whatever is coming out of this library.

It's not pretty, but you are on the right track.

Go ahead and merge the PR I gave you and keep going, I'll help you.

**Comment by rodrigc** [comment]
Hey @sideeffect42 , is the buildbot documentation for ldap correct?  I tried to follow it here:

http://docs.buildbot.net/latest/manual/cfg-www.html#user-information

but got an error starting up buildbot because I could have `url=c['buildbotURL'],` in the `c['www']` dict()

**Comment by sideeffect42** [comment]
@rodrigc Yeah, the docs are correct. I used them to set up my buildbot instance.

Not quite sure what you're trying to do.
In my master.cfg the setup is roughly like this:
```python
from buildbot.plugins import *

c = BuildmasterConfig = {
    'buildbotURL': "https://ci.example.com/",
    'buildbotNetUsageData': None,
}

userInfoProvider = util.LdapUserInfo(
    uri="ldap://ldap.example.com:389",
    bindUser="cn=buildbot,ou=dsa,dc=example,dc=com",
    bindPw="secret" 
    accountBase="ou=people,dc=example,dc=com",
    groupBase="ou=groups,dc=example,dc=com",
    accountPattern="(&(objectClass=inetOrgPerson)(uid=%(username)s))",
    accountFullName="cn",
    accountEmail="mail",
    groupMemberPattern="(&(objectClass=groupOfNames)(member=%(dn)s))",
    groupName="cn",
    avatarPattern="(&(objectClass=inetOrgPerson)(mail=%(email)s))",
    avatarData="jpegPhoto")
c['www'] = dict(
    port=8010,
    plugins=dict(waterfall_view={}, console_view={}, grid_view={}),
    auth=util.RemoteUserAuth(headerRegex="(?P<username>[^ @]+)", userInfoProvider=userInfoProvider),
    avatar_methods=[userInfoProvider])
```

**Pull request merged**